File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int left = 0;
        int right = 1;
        int max = 0;
        while(right < n){
            int buyprice = prices[left];
            int sellprice = prices[right];
            max = Math.max(max, sellprice-buyprice);
            if(buyprice > sellprice){
                left++;
            }
            else{
                right++;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1 
        right = max(piles)

        while left <= right:
            eating_speed = (left + right) // 2
            num_hours = 0
            for pile in piles:
                num_hours += math.ceil(pile / eating_speed) + 1
            # means you can decrease eating rate
            if num_hours <= h:
                right = eating_speed - 1
            else:
                left = eating_speed + 1
        return right